File: cpp/128_longest_consecutive_sequence.cpp
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
 public:
  int longestConsecutive(vector<int>& nums) {
    unordered_set<int> numbers(nums.begin(), nums.end());
    int result = 0;
    for (const auto& num : numbers) {
      if (numbers.count(num - 1) == 0) {
        int length = 1;
        while (numbers.count(num + length) > 0) {
          ++length;
        }
        result = max(result, length);
      }
    }
    return result;
  }
};

File: rust/128_longest_consecutive_sequence.rs
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        use std::collections::HashSet;

        let numbers: HashSet<i32> = nums.into_iter().collect();
        let mut result = 0;

        for num in &numbers {
            if !numbers.contains(&(num - 1)) {
                let mut length = 1;
                while numbers.contains(&(num + length)) {
                    length += 1;
                }
                result = std::cmp::max(result, length);
            }
        }

        result
    }
}
